spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -332,78 +423,62 @@ def _connect_control_signals(self):
+                    file = None
+        return file
 
     @property
-    def stderr_handle(self):
-        """Get handle to stderr_file."""
-        if self.stderr_file is not None:
-            # Needed to prevent any error that could appear.
-            # See spyder-ide/spyder#6267.
-            try:
-                handle = codecs.open(self.stderr_file, 'w', encoding='utf-8')
-            except Exception:
-                handle = None
-        else:
-            handle = None
-
-        return handle
+    def kernel_id(self):
+        """Get kernel id."""
+        if self.connection_file is not None:
+            json_file = osp.basename(self.connection_file)
+            return json_file.split('.json')[0]
 
-    def remove_stderr_file(self):
+    def remove_std_files(self, is_last_client=True):
         """Remove stderr_file associated with the client."""
         try:
-            # Defer closing the stderr_handle until the client
-            # is closed because jupyter_client needs it open
-            # while it tries to restart the kernel
-            self.stderr_handle.close()
-            os.remove(self.stderr_file)
-        except Exception:
+            self.shellwidget.executed.disconnect(self.poll_std_file_change)
+        except TypeError:
             pass
-
-    def get_stderr_contents(self):
-        """Get the contents of the stderr kernel file."""
-        try:
-            stderr = self._read_stderr()
-        except Exception:
-            stderr = None
-        return stderr
+        if self.std_poll_timer is not None:
+            self.std_poll_timer.stop()
+        if is_last_client:
+            if self.stderr_obj is not None:
+                self.stderr_obj.remove()
+            if self.stdout_obj is not None:
+                self.stdout_obj.remove()
+            if self.fault_obj is not None:
+                self.fault_obj.remove()
 
     @Slot()
-    def poll_stderr_file_change(self):
-        """Check if the stderr file just changed"""
-        try:
-            mtime = os.stat(self.stderr_file).st_mtime
-        except Exception:
-            return
-
-        if mtime == self.stderr_mtime:
-            return
-        self.stderr_mtime = mtime
-        stderr = self.get_stderr_contents()
+    def poll_std_file_change(self):
+        """Check if the stderr or stdout file just changed."""
+        self.shellwidget.call_kernel().flush_std()
+        stderr = self.stderr_obj.poll_file_change()
         if stderr:
+            if self.shellwidget.isHidden():
+                # Avoid printing the same thing again
+                if self.error_text != '<tt>%s</tt>' % stderr:
+                    full_stderr = self.stderr_obj.get_contents()
+                    self.show_kernel_error('<tt>%s</tt>' % full_stderr)
+            else:
+                self.shellwidget._append_plain_text(
+                    '\n' + stderr, before_prompt=True)
+        stdout = self.stdout_obj.poll_file_change()

```

Comment:
```suggestion

        stdout = self.stdout_obj.poll_file_change()
```

To visually separate stderr from stdout handling here.